File: app/src/main/java/com/gooner10/popularmoviesapp/Activity/ui/Activity/MovieActivity.java
package com.gooner10.popularmoviesapp.Activity.ui.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.GravityCompat;
import android.support.v4.view.ViewPager;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;

import com.gooner10.popularmoviesapp.Activity.domain.Model.MovieData;
import com.gooner10.popularmoviesapp.Activity.ui.adapter.MovieAdapter;
import com.gooner10.popularmoviesapp.Activity.ui.fragments.FavouriteFragment;
import com.gooner10.popularmoviesapp.Activity.ui.fragments.MovieFragment;
import com.gooner10.popularmoviesapp.R;

import java.util.ArrayList;

import butterknife.Bind;
import butterknife.ButterKnife;

public class MovieActivity extends AppCompatActivity {

    private final String LOG_TAG = "MovieActivity";

    private final Fragment mMovieFragment = new MovieFragment();
    private final Fragment mFavouriteFragment = new FavouriteFragment();
    private final ArrayList<MovieData> mMovieDataArrayList = new ArrayList<>();

    private String title;
    private String overview;
    private String poster_path;
    private String vote_average;
    private String release_date;
    private String backdrop_path;
    private String vote_count;
    private String popularity;

    @Bind(R.id.toolbar)
    Toolbar mToolbar;

    @Bind(R.id.drawer_layout)
    DrawerLayout mDrawerLayout;

    @Bind(R.id.viewpager)
    ViewPager mViewPager;

    @Bind(R.id.tabs)
    TabLayout mTabLayout;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_movie);

        // Bind all of the view
        ButterKnife.bind(this);

        // Setup Toolbar
        setUpToolbar();

        // Setup NavigationDrawer
        setupNavigationView();

        // Setup ViewPager
        setupViewPager();

        setUpHamburger();
        // Initialize Tabs
        mTabLayout = (TabLayout) findViewById(R.id.tabs);
        mTabLayout.setupWithViewPager(mViewPager);

    }

    private void setupViewPager() {
        if (mViewPager != null) {
            MovieAdapter mMovieAdapter = new MovieAdapter(getSupportFragmentManager());
            mMovieAdapter.addFragment(mMovieFragment, "Popular Movies");
            mMovieAdapter.addFragment(mFavouriteFragment, "Favourites");
            mViewPager.setAdapter(mMovieAdapter);
        }
    }

    private void setUpHamburger() {
        // Set up the hamburger icon to open and close the drawer
        ActionBarDrawerToggle mActionDrawerToggle = new ActionBarDrawerToggle(this, mDrawerLayout, mToolbar, R.string.drawer_open,
                R.string.drawer_close);
        mDrawerLayout.setDrawerListener(mActionDrawerToggle);
        mActionDrawerToggle.syncState();
    }

    private void setUpToolbar() {
        if (mToolbar != null) {
            setSupportActionBar(mToolbar);
        }
    }

    private void setupNavigationView() {
        if (mToolbar != null) {
            mToolbar.setNavigationIcon(R.drawable.ic_launcher);
            mToolbar.setNavigationOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mDrawerLayout.openDrawer(GravityCompat.START);
                    Log.d(LOG_TAG, "onClick");
                }
            });
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_movie, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            startActivity(new Intent(MovieActivity.this, SettingsActivity.class));
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
//        EventBus.getDefault().unregister(this);
    }
}

File: app/src/main/java/com/gooner10/popularmoviesapp/MovieActivity.kt
package com.gooner10.popularmoviesapp

import android.content.Intent
import android.os.Bundle
import android.preference.PreferenceManager
import android.util.Log
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.gooner10.moviehome.MovieFragment
import com.gooner10.popularmoviesapp.databinding.ActivityMovieBinding
import com.gooner10.popularmoviesapp.moviefavourite.FavouriteFragment
import com.gooner10.popularmoviesapp.trivia.MovieTriviaFragment
////import hugo.weaving.DebugLog
//import kotlinx.android.synthetic.main.activity_movie.*
//import kotlinx.android.synthetic.main.list_viewpager.*
//import kotlinx.android.synthetic.main.toolbar.*

class MovieActivity : AppCompatActivity() {
    private var movieFragment: Fragment? = null
    private var favouriteFragment: FavouriteFragment? = null
    private var triviaFragment: MovieTriviaFragment? = null
    private lateinit var binding: ActivityMovieBinding

//    //@DebugLog
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMovieBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        // Setup Toolbar
        setUpToolbar()

        // Setup NavigationDrawer
        setupNavigationView()

        // Setup ViewPager
        setupViewPager()

        setUpHamburger()
        // Initialize Tabs
        binding.viewpagerLayout.tabs.setupWithViewPager(binding.viewpagerLayout.viewpager)

//        PreferenceManager.setDefaultValues(this, R.xml.pref_general, false)
//        val sharedPref = PreferenceManager.getDefaultSharedPreferences(this)
//        val username = sharedPref.getString(getString(R.string.pref_sort_choice), "default value")
//        Log.d("TAG", username)
    }

    //@DebugLog
    private fun setupViewPager() {
//        Log.i(TAG, "setupViewPager: " + viewpager!!)
        val movieAdapter = MovieAdapter(supportFragmentManager)
        favouriteFragment = FavouriteFragment()
        movieFragment = MovieFragment()
        triviaFragment = MovieTriviaFragment()
        movieAdapter.addFragment(movieFragment as MovieFragment, "Popular")
        movieAdapter.addFragment(favouriteFragment!!, "Favourites")
        movieAdapter.addFragment(triviaFragment!!, "Trivia")
        Log.d(TAG, "setupViewPager favouriteFragment: " + favouriteFragment!!)
        binding.viewpagerLayout.viewpager.adapter = movieAdapter
        binding.viewpagerLayout.viewpager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
                Log.i(TAG, "onPageScrolled: $position")
            }

            override fun onPageSelected(position: Int) {
                Log.i(TAG, "onPageSelected: $position")
                if (position == 1) {
                    Log.d(TAG, "onPageSelected favouriteFragment: " + favouriteFragment!!)
                    favouriteFragment!!.setData()
                }
            }

            override fun onPageScrollStateChanged(state: Int) {
                Log.i(TAG, "onPageScrollStateChanged: $state")
            }
        })
    }

    private fun setUpHamburger() {
        // Set up the hamburger icon to open and close the drawer
//        val mActionDrawerToggle = ActionBarDrawerToggle(this, drawer_layout, toolbar, R.string.drawer_open,
//                R.string.drawer_close)
//        drawer_layout!!.setDrawerListener(mActionDrawerToggle)
//        mActionDrawerToggle.syncState()
    }

    private fun setUpToolbar() {
        setSupportActionBar(binding.viewpagerLayout.mainToolbar.toolbar)
    }

    private fun setupNavigationView() {
        val toolbar = binding.viewpagerLayout.mainToolbar.toolbar
        toolbar.setNavigationIcon(R.drawable.ic_launcher)
        toolbar.setNavigationOnClickListener {
//            drawer_layout!!.openDrawer(GravityCompat.START)
//            Log.d(TAG, "onClick")
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_movie, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId


        if (id == R.id.action_settings) {
            startActivity(Intent(this@MovieActivity, com.gooner10.settings.SettingsActivity::class.java))
        }

        return super.onOptionsItemSelected(item)
    }

    public override fun onDestroy() {
        super.onDestroy()
    }

    companion object {
        val TAG = MovieActivity::class.java.simpleName
    }
}
